utils/analytics.go

```diff
@@ -0,0 +1,62 @@
+package utils
+
+import (
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-security/utils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/xray/services"
+	xscservices "github.com/jfrog/jfrog-client-go/xsc/services"
+	"os"
+	"strings"
+)
+
+const (
+	githubActionsCi  = "github-actions"
+	jenkinsCi        = "jenkins"
+	gitlabCi         = "gitlab"
+	azurePipelinesCi = "azure-pipelines"
+)
+
+func AddAnalyticsGeneralEvent(gitInfoContext *services.XscGitInfoContext, serverDetails *config.ServerDetails, scanType string) *utils.AnalyticsMetricsService {
+	log.Debug("Initiating General Event report to Analytics service")
+	analyticsService := utils.NewAnalyticsMetricsService(serverDetails)
+	if !analyticsService.ShouldReportEvents() {
+		return analyticsService
+	}
+	analyticsGeneralEvent := createAnalyticsGeneralEvent(analyticsService, gitInfoContext, scanType)
+	analyticsService.AddGeneralEvent(analyticsGeneralEvent)
+	if analyticsService.GetMsi() != "" {
+		analyticsService.SetFinalizeEvent(&xscservices.XscAnalyticsGeneralEventFinalize{MultiScanId: analyticsService.GetMsi()})
+	} else {
+		analyticsService.SetShouldReportEvents(false)
+	}
+	return analyticsService
+}
+
+func createAnalyticsGeneralEvent(analyticsService *utils.AnalyticsMetricsService, gitInfo *services.XscGitInfoContext, scanType string) *xscservices.XscAnalyticsGeneralEvent {
+	generalEvent := analyticsService.CreateGeneralEvent(xscservices.FrogbotProduct, xscservices.FrogbotType)
+	generalEvent.ProductVersion = FrogbotVersion
+	generalEvent.FrogbotScanType = scanType
+	generalEvent.FrogbotCiProvider = resolveCi()
+	if gitInfo != nil {
+		generalEvent.GitInfo = gitInfo
+		generalEvent.IsGitInfoFlow = true
+	}
+	return generalEvent
+}
+
+func resolveCi() string {
+	switch {
+	case strings.ToLower(os.Getenv("GITHUB_ACTIONS")) == "true":
+		return githubActionsCi
+	case strings.ToLower(os.Getenv("GITLAB_CI")) == "true":
+		return gitlabCi
+	case os.Getenv("JENKINS_URL") != "":
+		return jenkinsCi
+	case strings.ToLower(os.Getenv("TF_BUILD")) == "true":
+		return azurePipelinesCi
+	// Currently, there isn't an environment variable specifically designed to identify JFrog Pipelines.
+	default:
+		return ""
+	}
+}
```

Comment:
* Add commnet to explain this.
* this may be implemented at `froggit-go` at `vcsutils/consts.go` for the `VcsProvider` enum